### PR TITLE
Roll up third_party/pigweed/repo to ToT

### PR DESCRIPTION
**Pull request by szatmz**
#### Problem
Recent Pigweed RPC changes are needed to support host-driven on-device testing.
Related PR: https://github.com/project-chip/connectedhomeip/pull/3930

 #### Summary of Changes
Rolling up third_party/pigweed/repo to a recent commit.

### CODE REVIEW

File: scripts/bootstrap.sh [hunk 1]
```diff
@@ -39,6 +39,9 @@ _bootstrap_or_activate() {
 EOF
     )"
 
+    PW_PROJECT_ROOT="$_CHIP_ROOT"
```
Comment (mspang) [inline scripts/bootstrap.sh:42]: What's this for?

Comment (szatmz) [reply scripts/bootstrap.sh:42]: Pigweed's virtualenv setup requires this be correctly set, for e.g. python modules outside of Pigweed repo be possible to install.

**Review by andy31415 — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: scripts/bootstrap.sh (hunk 1 above)
Comment (mspang) [reply scripts/bootstrap.sh:42]: I find the environment creep problematic, but it's not your fault so I'll take it up elsewhere.

**Comment by mspang** [comment]
Can you delete the old .environment in bootstrap?

**Review by mspang — CHANGES_REQUESTED** [review]
- We should delete the old .environment to save all developers 4G+ of space
- Lint fails. Please disable lint or fix the errors:

```
yuif:30320:~/connectedhomeip2% gn gen out/host
nDone. Made 267 targets from 104 files in 192ms
yuif:30321:~/connectedhomeip2% ninja -C out/host all
ninja: Entering directory `out/host'
[27/584] ACTION //integrations/mobly/py:py.wheel(//build/toolchain/host:linux_x64_gcc)
FAILED: gen/integrations/mobly/py/py.wheel.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/integrations/mobly/py --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --touch /home/spang/connectedhomeip2/out/host/gen/integrations/mobly/py/py.wheel.pw_pystamp --capture-output -- /home/spang/connectedhomeip2/third_party/pigweed/repo/pw_build/py/pw_build/python_wheels.py --out_dir /home/spang/connectedhomeip2/out/host/obj/integrations/mobly/py /home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/__init__.py /home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/pigweed_device.py /home/spang/connectedhomeip2/integrations/mobly/py/setup.py
Traceback (most recent call last):
  File "/home/spang/connectedhomeip2/third_party/pigweed/repo/pw_build/py/pw_build/python_wheels.py", line 64, in <module>
    main()
  File "/home/spang/connectedhomeip2/third_party/pigweed/repo/pw_build/py/pw_build/python_wheels.py", line 59, in main
    build_wheels(**vars(_parse_args()))
  File "/home/spang/connectedhomeip2/third_party/pigweed/repo/pw_build/py/pw_build/python_wheels.py", line 43, in build_wheels
    raise RuntimeError(f'Unable to find setup.py file at {filename}.')
RuntimeError: Unable to find setup.py file at /home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/__init__.py.
[36/584] ACTION //integrations/mobly/py:py.lint.pylint(//build/toolchain/host:linux_x64_gcc)
FAILED: gen/integrations/mobly/py/setup.py.pylint.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/integrations/mobly/py --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --directory /home/spang/connectedhomeip2/ --touch /home/spang/connectedhomeip2/out/host/gen/integrations/mobly/py/setup.py.pylint.pw_pystamp --capture-output --module pylint -- integrations/mobly/py/setup.py --jobs=1 --output-format=colorized
************* Module setup
integrations/mobly/py/setup.py:26:1: W0511: TODO - uncomment this once native python building is solved for (fixme)
integrations/mobly/py/setup.py:23:0: C0330: Wrong hanging indentation (add 4 spaces).
#    author_email='?',
^   | (bad-continuation)
integrations/mobly/py/setup.py:26:0: C0330: Wrong hanging indentation (add 4 spaces).
#TODO - uncomment this once native python building is solved for
^   | (bad-continuation)

--------------------------------------------------------------------
Your code has been rated at -5.00/10 (previous run: -5.00/10, +0.00)

[39/584] ACTION //integrations/mobly/py:py.lint.mypy(//build/toolchain/host:linux_x64_gcc)
FAILED: gen/integrations/mobly/py/py.lint.mypy.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/integrations/mobly/py --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --env=MYPY_FORCE_COLOR=1 --touch /home/spang/connectedhomeip2/out/host/gen/integrations/mobly/py/py.lint.mypy.pw_pystamp --capture-output --module mypy -- --pretty --show-error-codes /home/spang/connectedhomeip2/integrations/mobly/py
/home/spang/connectedhomeip2/integrations/mobly/py/setup.py:17: error: Skipping
analyzing 'setuptools': found module but no type hints or library stubs 
[import]
    import setuptools
    ^
/home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/pigweed_device.py:17: error:
Skipping analyzing 'serial': found module but no type hints or library stubs 
[import]
    import serial
    ^
/home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/pigweed_device.py:17: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
/home/spang/connectedhomeip2/integrations/mobly/py/chip_mobly/pigweed_device.py:20: error:
Skipping analyzing 'pw_hdlc_lite.rpc': found module but no type hints or library
stubs  [import]
    from pw_hdlc_lite.rpc import HdlcRpcClient
    ^
/home/spang/connectedhomeip2/integrations/mobly/py/hello_world_test.py:16: error:
Skipping analyzing 'mobly': found module but no type hints or library stubs 
[import]
    from mobly import asserts
    ^
Found 4 errors in 3 files (checked 4 source files)
[40/584] ACTION //integrations/mobly/py:py.lint.pylint(//build/toolchain/host:linux_x64_gcc)
FAILED: gen/integrations/mobly/py/chip_mobly/pigweed_device.py.pylint.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/integrations/mobly/py --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --directory /home/spang/connectedhomeip2/ --touch /home/spang/connectedhomeip2/out/host/gen/integrations/mobly/py/chip_mobly/pigweed_device.py.pylint.pw_pystamp --capture-output --module pylint -- integrations/mobly/py/chip_mobly/pigweed_device.py --jobs=1 --output-format=colorized
************* Module chip_mobly.pigweed_device
integrations/mobly/py/chip_mobly/pigweed_device.py:1:0: C0114: Missing module docstring (missing-module-docstring)
integrations/mobly/py/chip_mobly/pigweed_device.py:32:0: C0115: Missing class docstring (missing-class-docstring)
integrations/mobly/py/chip_mobly/pigweed_device.py:39:12: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
integrations/mobly/py/chip_mobly/pigweed_device.py:43:4: C0116: Missing function or method docstring (missing-function-docstring)
integrations/mobly/py/chip_mobly/pigweed_device.py:47:4: C0116: Missing function or method docstring (missing-function-docstring)
integrations/mobly/py/chip_mobly/pigweed_device.py:74:4: W0107: Unnecessary pass statement (unnecessary-pass)
integrations/mobly/py/chip_mobly/pigweed_device.py:18:0: C0411: standard import "import importlib" should be placed before "import serial" (wrong-import-order)

------------------------------------------------------------------
Your code has been rated at 8.06/10 (previous run: 8.06/10, +0.00)

ninja: build stopped: subcommand failed.

```

**Comment by szatmz** [comment]
@mspang please try now

**Comment by mspang** [comment]
> @mspang please try now

```
FAILED: gen/third_party/happy/repo/setup.py.pylint.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/third_party/happy --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --directory /home/spang/connectedhomeip2/ --touch /home/spang/connectedhomeip2/out/host/gen/third_party/happy/repo/setup.py.pylint.pw_pystamp --capture-output --module pylint -- third_party/happy/repo/setup.py --jobs=1 --output-format=colorized
************* Module setup
third_party/happy/repo/setup.py:1:0: C0114: Missing module docstring (missing-module-docstring)
third_party/happy/repo/setup.py:21:0: W0122: Use of exec (exec-used)
third_party/happy/repo/setup.py:28:14: E0602: Undefined variable '__version__' (undefined-variable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)

[78/178] ACTION //third_party/happy:happy.lint.mypy(//build/toolchain/host:linux_x64_gcc)
FAILED: gen/third_party/happy/happy.lint.mypy.pw_pystamp 
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /home/spang/connectedhomeip2/ --current-path /home/spang/connectedhomeip2/third_party/happy --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//build/toolchain/host:linux_x64_gcc --env=MYPY_FORCE_COLOR=1 --touch /home/spang/connectedhomeip2/out/host/gen/third_party/happy/happy.lint.mypy.pw_pystamp --capture-output --module mypy -- --pretty --show-error-codes /home/spang/connectedhomeip2/third_party/happy/repo
/home/spang/connectedhomeip2/third_party/happy/repo/setup.py:19: error:
Skipping analyzing 'setuptools': found module but no type hints or library stubs
 [import]
    from setuptools import find_packages, setup
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/setup.py:28: error: Name
'__version__' is not defined  [name-defined]
          version=__version__,
                  ^
/home/spang/connectedhomeip2/third_party/happy/repo/pip_packages.py:27: error:
Skipping analyzing 'pip': found module but no type hints or library stubs 
[import]
    import pip
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/plugins/testrail/TestrailResultOutput.py:32: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["file_path"] = None
                           ^
/home/spang/connectedhomeip2/third_party/happy/repo/plugins/testrail/TestrailResultOutput.py:33: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["output_data"] = None
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:44: error:
Need type annotation for 'options' (hint: "options: Dict[<type>, <type>] = ...")
 [var-annotated]
    options = {}
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:47: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options['action'] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:48: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options['ips'] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:49: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options['dstPort'] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:50: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options['start'] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:51: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options['duration'] = None
                          ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:511: error:
Incompatible types in assignment (expression has type "str", target has type
"bool")  [assignment]
                options["interface"] = a
                                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:514: error:
Incompatible types in assignment (expression has type "str", target has type
"bool")  [assignment]
                options["action"] = a
                                    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:517: error:
Incompatible types in assignment (expression has type "List[str]", target has
type "bool")  [assignment]
                options["ips"] = a.split(',')
                                 ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:520: error:
Incompatible types in assignment (expression has type "int", target has type
"bool")  [assignment]
                options["dstPort"] = int(a)
                                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:523: error:
Incompatible types in assignment (expression has type "int", target has type
"bool")  [assignment]
                options["start"] = int(a)
                                   ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:526: error:
Incompatible types in assignment (expression has type "int", target has type
"bool")  [assignment]
                options["duration"] = int(a)
                                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyPacketProcess.py:554: error:
Need type annotation for 'packetList' (hint: "packetList: List[<type>] = ...") 
[var-annotated]
        packetList = manager.list()
        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Driver.py:31: error:
Skipping analyzing 'lockfile': found module but no type hints or library stubs 
[import]
    import lockfile
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Driver.py:31: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
/home/spang/connectedhomeip2/third_party/happy/repo/happy/State.py:257: error:
Name 'getNodeInterfaceAddressMask' already defined on line 212  [no-redef]
        def getNodeInterfaceAddressMask(self, interface_id, addr, node_id=...
        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyConfiguration.py:36: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["delete"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyConfiguration.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["key"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyConfiguration.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["value"] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyConfiguration.py:39: error:
Incompatible types in assignment (expression has type "str", target has type
"bool")  [assignment]
    options["config-type"] = "user"
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyLinkDelete.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["link_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcess.py:27: error:
Skipping analyzing 'psutil': found module but no type hints or library stubs 
[import]
    import psutil
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyLinkAdd.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["type"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Traceroute.py:36: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["source"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Traceroute.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["destination"] = None
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Ping.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["source"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Ping.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["destination"] = None
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Ping.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["size"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/Ping.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["count"] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyShell.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyShell.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["command"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessWait.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessWait.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["tag"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessWait.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["timeout"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStrace.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStrace.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["tag"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStop.py:32: error:
Skipping analyzing 'psutil': found module but no type hints or library stubs 
[import]
    import psutil
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStop.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStop.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["tag"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessOutput.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessOutput.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["tag"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTmux.py:36: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTmux.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["run_as_user"] = None
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTmux.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["session"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeStatus.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeLeave.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeLeave.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeAddress.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeAddress.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["interface"] = None
                           ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeAddress.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["address"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkStatus.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkState.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyDNS.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["dns"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyDNS.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:39: error:
Cannot find implementation or library stub for module named 'networkx'  [import]
        import networkx as nx
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:45: error:
Cannot find implementation or library stub for module named 'matplotlib.pyplot' 
[import]
        import matplotlib.pyplot as plt
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:45: error:
Cannot find implementation or library stub for module named 'matplotlib' 
[import]
        import matplotlib.pyplot as plt
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:52: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["save"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:53: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["graph"] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:59: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyState.py:60: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["extension"] = None
                           ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:33: error:
Skipping analyzing 'psutil': found module but no type hints or library stubs 
[import]
    import psutil
    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:45: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["tag"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:46: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["command"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:48: error:
Incompatible types in assignment (expression has type
"Dict[<nothing>, <nothing>]", target has type "bool")  [assignment]
    options["env"] = {}
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyProcessStart.py:49: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["sync_on_output"] = None
                                ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeDelete.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkDelete.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkAddress.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkAddress.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["address"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyStateUnload.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["json_file"] = None
                           ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:37: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:38: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["action"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:39: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["interface"] = None
                           ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["start"] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["duration"] = None
                          ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["ips"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeTcpReset.py:43: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["dstPort"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:45: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["to"] = None
                    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:46: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["via"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:47: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["prefix"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:49: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["isp"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:50: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["seed"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeRoute.py:51: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["route_type"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeAdd.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeAdd.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["type"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkAdd.py:42: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkAdd.py:43: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["type"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeJoin.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeJoin.py:46: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeJoin.py:47: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["fix_hw_addr"] = None
                             ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNodeJoin.py:48: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["customized_eui64"] = None
                                  ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["network_id"] = None
                            ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["to"] = None
                    ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:45: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["via"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:46: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["prefix"] = None
                        ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:48: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["isp"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyNetworkRoute.py:49: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["seed"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyInternet.py:40: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["node_id"] = None
                         ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyInternet.py:41: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["iface"] = None
                       ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyInternet.py:44: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["isp"] = None
                     ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyInternet.py:45: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["seed"] = None
                      ^
/home/spang/connectedhomeip2/third_party/happy/repo/happy/HappyStateLoad.py:45: error:
Incompatible types in assignment (expression has type "None", target has type
"bool")  [assignment]
    options["json_file"] = None
                           ^
Found 109 errors in 38 files (checked 61 source files)
[81/178] ACTION //src/lib/core/tests:TestCHIPTLV_run(//build/toolchain/host:linux_x64_gcc)
ninja: build stopped: subcommand failed.
```

**Comment by szatmz** [comment]
@mspang try again please

**Review by mspang — APPROVED** [review]

**Comment by mspang** [comment]
@saurabhst @hawk248 @BroderickCarlin @chrisdecenzo @jelderton

**Review by Damian-Nordic — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Pull request merged**